src/cargo/util/config.rs

```diff
@@ -1784,3 +1784,13 @@ impl Drop for PackageCacheLock<'_> {
         }
     }
 }
+
+/// returns path to clippy-driver binary
+///
+/// Allows override of the path via `CARGO_CLIPPY_DRIVER` env variable
+pub fn clippy_driver() -> PathBuf {
+    env::var("CARGO_CLIPPY_DRIVER")
+        .ok()
```

Comment:
This `.ok` isn't necessary.  Can just change `unwrap_or_else` to use `|_|`.

Reply:
out of curiosity, do you think it optimizes to different code either way?

Reply:
I would guess that it does.  I just find it a little more straightforward to read instead of swapping between types.